crates/re_sdk/src/spawn.rs

```diff
@@ -0,0 +1,166 @@
+        const RERUN_BINARY: &str = "rerun";
+
+        if let Some(path) = self.executable_path.as_deref() {
+            return path.to_owned();
+        }
+
+        self.executable_name
+            .as_deref()
+            .unwrap_or(RERUN_BINARY)
+            .to_owned()
+    }
+}
+
+/// Spawns a new Rerun Viewer process ready to listen for TCP connections.
+///
+/// Refer to [`SpawnOptions`]'s documentation for configuration options.
+///
+/// This only starts a Viewer process: if you'd like to connect to it and start sending data, refer
+/// to [`crate::RecordingStream::connect`] or use [`crate::RecordingStream::spawn`] directly.
+pub fn spawn(opts: &SpawnOptions) -> std::io::Result<()> {
+    use std::{net::TcpStream, process::Command, time::Duration};
+
+    // NOTE: It's indented on purpose, it just looks better and reads easier.
+    const EXECUTABLE_NOT_FOUND: &str = //
+    "
+    Couldn't find the Rerun Viewer executable in your PATH.
+
+    You can install binary releases of the Rerun Viewer using any of the following methods:
+    * Binary download with `cargo`: `cargo binstall rerun-cli` (see https://github.com/cargo-bins/cargo-binstall)
+    * Build from source with `cargo`: `cargo install rerun-cli` (requires Rust 1.72+)
+    * Direct download from our release assets: https://github.com/rerun-io/rerun/releases/latest/
+    * Or together with the Rerun Python SDK:
+      * Pip: `pip3 install rerun-sdk`
+      * Conda: `conda install -c conda-forge rerun-sdk`
+      * Binary download with `pixi`: `pixi global install rerun-sdk` (see https://prefix.dev/docs/pixi/overview)
+
+    If your platform and/or architecture is not available, you can refer to
+    https://github.com/rerun-io/rerun/blob/main/BUILD.md for instructions on how to build from source.
+
+    Otherwise, feel free to open an issue at https://github.com/rerun-io/rerun/issues if you'd like to
+    request binary releases for your specific platform.
+    ";
+
+    let port = opts.port();
+    let connect_addr = opts.connect_addr();
+    let memory_limit = opts.memory_limit();
+    let executable_path = opts.executable_path();
+
+    if TcpStream::connect_timeout(&connect_addr, Duration::from_millis(1)).is_ok() {
+        re_log::info!(
+            addr = %opts.listen_addr(),
+            "A process is already listening at this address. Assuming it's a Rerun Viewer."
+        );
+        return Ok(());
+    }
+
+    let res = Command::new(executable_path)
+        .arg(format!("--port={port}"))
+        .arg(format!("--memory-limit={memory_limit}"))
+        .arg("--skip-welcome-screen")
+        .spawn();
+
+    let rerun_bin = match res {
+        Ok(rerun_bin) => rerun_bin,
+        Err(err) if err.kind() == std::io::ErrorKind::NotFound => {
+            eprintln!("{EXECUTABLE_NOT_FOUND}");
+            return Err(err);
+        }
+        Err(err) => {
+            re_log::info!(%err, "Failed to spawn Rerun Viewer");
+            return Err(err);
```

Comment:
Again - never do this. Errors should EITHER be shown to the user (logged) OR returned, but not both:

https://github.com/rerun-io/rerun/blob/main/CODE_STYLE.md#error-handling-and-logging

Reply:
I'm breaking the rule because I need to print something on the Rust side, since I cant/dont want to pass all of this to C and C++, while still letting them know that an error occured...

I could do the print at the last minute in `rerun_c`, that would be an improvement I think.

Reply:
I'll see how painful that is